### PR TITLE
WIP Feature/elasticsearch upgrade

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
Fixes # (link to Jira or GitHub issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] I have updated the [CHANGELOG](../CHANGELOG.md)
- [ ] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/Http/Controllers/CatalogController.php [hunk 1]
```diff
@@ -83,71 +87,61 @@ public function getIndex()
...
-
-            $params['query']['bool']['filter'] = Item::getFilterParams($filter);
-            if (!empty($input['year-range']) &&
-                $input['year-range'] != Item::sliderMin().','.Item::sliderMax() //nezmenena hodnota
-            ) {
-                $range = explode(',', $input['year-range']);
-                $params['query']['bool']['filter']['and'][]['range']['date_earliest']['gte'] = (isset($range[0])) ? $range[0] : Item::sliderMin();
-                $params['query']['bool']['filter']['and'][]['range']['date_latest']['lte'] = (isset($range[1])) ? $range[1] : Item::sliderMax();
-            }
+            $params['query']['bool']['should'] = $should;
```
Comment (martin-havala) [inline app/Http/Controllers/CatalogController.php]: Are $params still used? 
I noticed declaration has been removed up there and they're not used anywhere

File: app/Http/routes.php [hunk 2]
```diff
@@ -41,14 +43,32 @@ function()
...
     });
 
-    Route::get('/', function () {
+    Route::get('/', function (
+        AuthorityRepository $authorityRepository,
+        ItemRepository $itemRepository
+    ) {
+        $choices = [
+            ['intro.from_galleries_start', 'intro.from_galleries_end', route('frontend.info'), 9],
+            ['intro.from_galleries_start', 'intro.from_authors_end', route('frontend.author.index'), $authorityRepository->count()],
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app/Http/routes.php]: thanks!

File: app/Http/Controllers/CatalogController.php (hunk 1 above)
Comment (rastislav-chynoransky) [reply app/Http/Controllers/CatalogController.php]: this PR is WIP. the method will be refactored

Comment (igor-kamil) [reply app/Http/Controllers/CatalogController.php]: btw - next time you can open it as "draft PR" - nice new feature - https://github.blog/2019-02-14-introducing-draft-pull-requests/

File: app/Constraints/Range.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace App\Constraints;
+
+use Symfony\Component\Validator\Constraints\Range as BaseRange;
+
+class Range extends BaseRange
```
Comment (martin-havala) [inline app/Constraints/Range.php]: is it used anywhere?

**Pull request closed without merging**